daily_milestone.go

```diff
@@ -1,29 +1,120 @@
 package main
 
-import "flag"
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
+// Struct to be used for milestone
+// Milestone ....
+type gitLabAPI struct {
+	ID          int               `json:"id"`
+	Iid         int               `json:"iid"`
+	ProjectID   int               `json:"project_id"`
+	Title       string            `json:"title"`
+	Description string            `json:"description"`
+	StartDate   string            `json:"start_date"`
+	DueDate     string            `json:"due_date"`
+	State       string            `json:"state"`
+	UpdatedAt   *time.Time        `json:"updated_at"`
+	CreatedAt   *time.Time        `json:"created_at"`
+	Name        string            `json:"name"`
+	NameSpace   map[string]string `json:"namespace"`
+}
+
+// Initialization of logging variable
+var logger *log.Logger
+
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, Token string, Projectname string, Namespace string) string {
+	project := gitLabAPI{}
+	urls := "https://" + baseURL + "/projects"
 	page := 1
+	strPage := strconv.Itoa(page)
+	s := []string{urls, "?page=", strPage}
+	completeURL := strings.Join(s, "")
 	for {
-
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", completeURL, nil)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		req.Header.Add("Token", Token)
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		respByte, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			logger.Println("fail to read response data")
+			break
+		}
+		json.Unmarshal(respByte, project)
+		defer resp.Body.Close()
+		fmt.Println(resp.Body)
+		fmt.Println(project)
+		if project.Name == "message" {
+			fmt.Println(project.Name)
+		}
+		if project.Name == Projectname && project.NameSpace["path"] == Namespace {
+			return strconv.Itoa(project.ID)
+		}
+		if project.Title == "" {
+			break
+		}
+		page++
```

Comment:
Hmmm currently this variable isn't reused anywhere so incrementing it does nothing